packages/gensx-core/src/component.ts

```diff
@@ -62,62 +61,11 @@ function getResolvedOpts(
   return merged;
 }
 
-export function Component(decoratorOpts?: DecoratorComponentOpts) {
-  return function <P extends object = {}, R = unknown>(
-    target: (props: P) => R,
-    context?:
-      | ClassMethodDecoratorContext
-      | ClassAccessorDecoratorContext
-      | ClassFieldDecoratorContext
-      | ClassDecoratorContext,
-  ): (props?: P) => R {
-    // Only wrap class methods
-    if (context && context.kind !== "method") {
-      console.warn("Component decorator can only be applied to class methods.");
-      return target as (props?: P) => R;
-    }
-
-    return createComponent<P, R>(target, decoratorOpts) as (props?: P) => R;
-  };
-}
-
-export function Workflow(decoratorOpts?: DecoratorWorkflowOpts) {
-  return function <P extends object = {}, R = unknown>(
-    target: (props: P) => R,
-    context?:
-      | ClassMethodDecoratorContext
-      | ClassAccessorDecoratorContext
-      | ClassDecoratorContext
-      | ClassFieldDecoratorContext,
-  ): (props?: P) => Promise<Awaited<R>> {
-    // Only wrap class methods
-    if (context && context.kind !== "method") {
-      console.warn("Workflow decorator can only be applied to class methods.");
-      return (async (props?: P) => await target((props ?? {}) as P)) as (
-        props?: P,
-      ) => Promise<Awaited<R>>;
-    }
-
-    return createWorkflow<P, R>(target, decoratorOpts);
-  };
-}
-
-export function createComponent<P extends object = {}, R = unknown>(
+export function Component<P extends object = {}, R = unknown>(
+  name: string,
   target: (props: P) => R,
```

Comment:
Is it possible to widen this type? Accept any function with any shape of args?

I guess we would need to change the tracing API...